src/script/script_instance.hpp

```diff
@@ -283,10 +282,20 @@ class ScriptInstance {
 	 */
 	virtual void LoadDummyScript() = 0;
 
+	/**
+	 * Get the storage of this script.
+	 */
+	inline class Squirrel &GetEngine()
+	{
+		assert(this->engine != nullptr);
+		return *this->engine;
+	}
+
 private:
-	class ScriptController *controller = nullptr; ///< The script main class.
-	class ScriptStorage *storage = nullptr; ///< Some global information for each running script.
-	SQObject *instance = nullptr; ///< Squirrel-pointer to the script main class.
+	std::unique_ptr<class ScriptStorage> storage; ///< Some global information for each running script.
+	std::unique_ptr<class Squirrel> engine; ///< A wrapper around the squirrel vm.
+	std::unique_ptr<class ScriptController> controller; ///< The script main class.
+	std::unique_ptr<SQObject> instance; ///< Squirrel-pointer to the script main class.
```

Comment:
As `~engine()` requires instance to exist, switch the order of these up to ensure correct destruction order.